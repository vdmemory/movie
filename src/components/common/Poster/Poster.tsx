import React from "react";

import noposter from "./noposter.png";

type Props = {
    title: string;
    poster: string;
};

const Card: React.FC<Props> = ({ title, poster }) => {
    return <img alt={title} src={poster !== "N/A" ? poster : noposter} />;
};

export default Card;
