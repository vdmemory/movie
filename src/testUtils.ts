export const deepEqual = (a: any, b: any) => {
    if (a === b) {
        return true;
    }

    if (
        a == null ||
        typeof a != "object" ||
        b == null ||
        typeof b != "object"
    ) {
        return false;
    }

    var propertiesInA = 0,
        propertiesInB = 0;
    for (var property in a) {
        propertiesInA += 1;
    }
    for (var property in b) {
        propertiesInB += 1;
        if (!(property in a) || !deepEqual(a[property], b[property])) {
            return false;
        }
    }
    return propertiesInA == propertiesInB;
};
