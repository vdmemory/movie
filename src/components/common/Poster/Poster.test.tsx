import React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import Poster from "./Poster";
import { deepEqual } from "../../../testUtils";

type Props = {
    title: string;
    poster: string;
};

const setup = (props: Props) => {
    const renderer = createRenderer();
    renderer.render(<Poster {...props} />);
    const output = renderer.getRenderOutput();
    return { output, props };
};

const propsRender: Props = {
    title: "Title",
    poster: "https://poster.jpg",
};

const propsNotRender: Props = {
    title: "Title",
    poster: "N/A",
};

describe("Poster Component", () => {
    describe("Render component and its components", () => {
        test("Should renders correctly", () => {
            const { output } = setup(propsRender);
            expect(output).toMatchSnapshot();
        });

        test("Should render image", () => {
            const { output } = setup(propsRender);
            expect(output.type).toBe("img");
        });

        test("Should render the filled alt attribute and src attribute", () => {
            const { output } = setup(propsRender);
            expect(output.props.alt).toBe("Title");
            expect(output.props.src).toBe("https://poster.jpg");
        });
        
        test("Should display the plug if props poster 'N/A' in src attribute value", () => {
            const { output } = setup(propsNotRender);
            expect(output.props.src).toBe("noposter.png");
        });

        describe("Check props", () => {
            const expectedProps = {
                title: "Title",
                poster: "https://poster.jpg",
            };
            test("Should render and check props", () => {
                const { props } = setup(propsRender);
                expect(deepEqual(props, expectedProps)).toBeTruthy();
            });
        });
    });
});
