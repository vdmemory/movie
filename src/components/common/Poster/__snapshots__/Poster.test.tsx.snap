// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster Component Render component and its components Should renders correctly 1`] = `
<img
  alt="Title"
  src="https://poster.jpg"
/>
`;
